### ISSUE TITLE
Regression: Path.contains_points now returns uint instead of bool

### ISSUE THREAD

**Issue by astrofrog** [matplotlib/matplotlib#6566]
In Matplotlib 1.5.1, contains_points return a boolean, consistent with the docstring:

```python
In [1]: from matplotlib.path import Path

In [2]: Path([[1,2],[3,4],[5,6]]).contains_points([[1,2]])
Out[2]: array([False], dtype=bool)
```

However, in the latest developer version, an ``uint8`` is returned:

```python
In [15]: from matplotlib.path import Path

In [16]: Path([[1,2],[3,4],[5,6]]).contains_points([[1,2]])
Out[16]: array([0], dtype=uint8)
```

This causes issues with packages that are assuming bool, since masking arrays using an array of ints has a different results (it will either select the first or second element or both from an array), so I believe this is a regression.

**Comment by tacaswell** [comment]
In < 1.4 it returned uints, 1.5 return bool (as part of removing CXX usage), and we reverted to uint in https://github.com/matplotlib/matplotlib/pull/6450 as part of fixing a major path related speed regression.

We should probably cast to bool on the way out?

attn @mdboom

**Comment by WeatherGod** [comment]
I thought we were casting to bool on the way out? Is that only on master
then?

**Comment by astrofrog** [comment]
@WeatherGod - this is on master, I haven't tested elsewhere

**Comment by tacaswell** [comment]
attn @mdboom

**Comment by tacaswell** [comment]
I think we should revert the change to return uint -> bool.  We broke that api 1.4->1.5 without realizing it, but it is better to stick with it (rather than swing back and forth).

Also as @astrofrog the bool is more natural to use for masking.

**Comment by jenshnielsen** [comment]
Closed by #6654
